src/display/annotation_layer.js

```diff
@@ -676,7 +678,8 @@ class TextAnnotationElement extends AnnotationElement {
     const isRenderable = !!(
       parameters.data.hasPopup ||
       parameters.data.titleObj?.str ||
-      parameters.data.contentsObj?.str
+      parameters.data.contentsObj?.str ||
+      parameters.data.richText
```

Comment:
Given my other comments, I suppose all of these newly added checks should perhaps be `parameters.data.richText?.str` instead?